src/components/CreateGroup/CreateGroupForm.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react'
+import Typography from '@material-ui/core/Typography'
+import Box from '@material-ui/core/Box'
+import { TextField } from 'formik-material-ui'
+import { spacing } from '@material-ui/system'
+import { styled } from '@material-ui/core/styles'
+import MuiButton from '@material-ui/core/Button'
+import { Formik, Form, Field } from 'formik'
+import CheckBoxIcon from '@material-ui/icons/CheckBox'
+import { useTranslation } from 'react-i18next'
+import { selectToken } from '../App/userSelector'
+import { useSelector } from 'react-redux'
+import { createGroup } from '../../pages/AdminDashboard/groupsActions'
+import { useDispatch } from 'react-redux'
+
+const Button = styled(MuiButton)(spacing)
+
+export interface Group {
+    name: string
+    password: string
+}
+
+const CreateGroupForm = () => {
+    const { t } = useTranslation('create-group-form')
+    const token = useSelector(selectToken)
+    const dispatch = useDispatch()
+
+
+    const validateForm = (values) => {
+        const errors: Partial<Group> = {}
+        const requiredFields = ['name', 'password']
+        for (const requiredField of requiredFields) {
+            if (!values[requiredField]) {
+                errors[requiredField] = 'Required'
+            }
+        }
+        return errors
+    }
```

Comment:
pls consider using `yup` for form validation. There should already be an example within the code